tests/DCPS/LivelinessTest/dualpub.cpp

```diff
@@ -0,0 +1,379 @@
+// -*- C++ -*-
```

Comment:
It looks like you copied from another test.  Which is fine.  However, I would seriously consider re-writing a single-process test if possible and look for a way to eliminate as many sleeps as possible.